File: app/src/main/java/com/xpayworld/payment/di/module/ActivityModule.java
package com.xpayworld.payment.di.module;

import android.content.Context;
import androidx.appcompat.app.AppCompatActivity;
import com.xpayworld.payment.di.ActivityContext;
import dagger.Module;
import dagger.Provides;


@Module
public class ActivityModule {
    private AppCompatActivity mActivity;

    public ActivityModule(AppCompatActivity activity) {
        this.mActivity = activity;
    }

    @Provides
    @ActivityContext
    Context provideContext() {
        return mActivity;
    }
    @Provides
    AppCompatActivity provideActivity() {
        return mActivity;
    }
}

File: app/src/main/java/com/xpayworld/payment/di/module/ActivityModule.kt
package com.xpayworld.payment.di.module

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import com.xpayworld.payment.di.ActivityContext
import dagger.Module
import dagger.Provides


@Module
class ActivityModule(private val mActivity: AppCompatActivity) {

    @Provides
    @ActivityContext
    internal fun provideContext(): Context {
        return mActivity
    }

    @Provides
    internal fun provideActivity(): AppCompatActivity {
        return mActivity
    }
}
